pkg/operator/ceph/disruption/clusterdisruption/osd.go

```diff
@@ -53,15 +54,31 @@ const (
 	nooutFlag                 = "noout"
 )
 
-func (r *ReconcileClusterDisruption) createPDB(pdb *policyv1beta1.PodDisruptionBudget) error {
+func (r *ReconcileClusterDisruption) createPDBV1(pdb *policyv1.PodDisruptionBudget) error {
 	err := r.client.Create(context.TODO(), pdb)
 	if err != nil && !apierrors.IsAlreadyExists(err) {
 		return errors.Wrapf(err, "failed to create pdb %q", pdb.Name)
 	}
 	return nil
 }
 
-func (r *ReconcileClusterDisruption) deletePDB(pdb *policyv1beta1.PodDisruptionBudget) error {
+func (r *ReconcileClusterDisruption) createPDBV1beta1(pdb *policyv1beta1.PodDisruptionBudget) error {
+	err := r.client.Create(context.TODO(), pdb)
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		return errors.Wrapf(err, "failed to create pdb %q", pdb.Name)
+	}
+	return nil
+}
+
+func (r *ReconcileClusterDisruption) deletePDBV1(pdb *policyv1.PodDisruptionBudget) error {
```

Comment:
I don't think you need separate methods for deleting v1 or v1beta1 objects. The `r.client.Delete()` takes a `client.Object` type, so what about defining a method like this:
```suggestion
func (r *ReconcileClusterDisruption) deletePDB(pdb client.Object) error {
```

Reply:
Updated